postStart.sh

```diff
@@ -56,30 +51,37 @@ function setPasswordFromFile() {
 
 if curl --fail --silent -u "${USERNAME}":"${PASSWORD}" http://"${HOST}"/service/metrics/ping; then
```

Comment:
@deanesmith, have you tested this? I am pretty sure that this won't work anymore. Nexus does not use the default 'admin123' password anymore. From the [release notes](https://help.sonatype.com/repomanager3/release-notes/2019-release-notes):

> Default admin password
>
> You will find that the default password for the admin user is now randomly generated on startup of a new NXRM  instance; see the Post Install Checklist for more details.

I did the following:

```sh
> docker run -p 8081:8081 sonatype/nexus3:3.22.1
```

in another terminal:

```sh
> curl --fail  -u "admin:admin123" http://localhost:8081/service/metrics/ping
curl: (22) The requested URL returned error: 401

> docker ps
CONTAINER ID        IMAGE                    COMMAND                  CREATED             STATUS              PORTS                    NAMES
2a51229a3f83        sonatype/nexus3:3.22.1   "sh -c ${SONATYPE_DI…"   17 minutes ago      Up 17 minutes       0.0.0.0:8081->8081/tcp   festive_lehmann

> docker exec -it 2a51229a3f83 cat /nexus-data/admin.password
e867a834-cca1-4709-bdf0-7a3c9abdc134

> curl --fail  -u "admin:e867a834-cca1-4709-bdf0-7a3c9abdc134" http://localhost:8081/service/metrics/ping
pong
```

Bottom line, I think one needs to retrieve the initial admin password from the container itself.

Reply:
It is set here I believe - https://github.com/jenkins-x-charts/nexus/blob/master/nexus/templates/secret.yaml

Reply:
that just creates a secret, but there seems to be a magic env variable https://github.com/jenkins-x-charts/nexus/blob/master/nexus/values.yaml#L25

Reply:
Ok, so it seems NEXUS_SECURITY_RANDOMPASSWORD prevents the random password.